### PR TITLE
Automatically add a namespace prefix to the HTTP controller

### PR DESCRIPTION
**Pull request by lilianjin**

### CODE REVIEW

**Comment by lilianjin** [comment]
All along, the Pyro routing configuration is written in a full name, which results in a long name. It is expected that the method of constructing the short circuit will improve the efficiency of reading and development.

File: src/Addon/AddonServiceProvider.php [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?php namespace Anomaly\Streams\Platform\Addon;
 
+use ReflectionClass;
```
Comment (Piterden) [inline src/Addon/AddonServiceProvider.php:3]: Laravel has the representation of this stuff, as I remember...

Try `app(get_class($this))` or `app()->make(get_class($this))`.

BTW, why not `static::class`? (not tested it!)

File: src/Addon/AddonServiceProvider.php [hunk 2]
```diff
@@ -240,6 +241,21 @@ public function getListeners()
...
+	 * In addition, it is set as the URL generator's root namespace.
+	 *
+	 * @param  string $name
+	 * @return string
+	 */
+	public function getHttpNamepsace($name = '')
+	{
+		$name = $name !== '' ?  trim($name, '\\') : $name;
+		return (new ReflectionClass(get_class($this)))
+			->getNamespaceName() . '\\Http\\Controller\\' . $name;
```
Comment (Piterden) [inline src/Addon/AddonServiceProvider.php:257]: Read more sources, my friend )))

**Comment by RyanThompson** [comment]
Are you saying this will make a marked improvement in performance? If so can you share your testing results? Otherwise I am keen to keep it the way it is.

**Comment by RyanThompson** [comment]
I think I am going to close this for now. The reflection has always bothered me. It tends to be expensive and we do plenty of it already in the IoC container. Let's keep the concept in mind though. 4.0 will automate routing so maybe this will be less needed anyways!

**Pull request closed without merging**